ext/mysqlnd/mysqlnd_charset.c

```diff
@@ -840,8 +840,10 @@ PHPAPI zend_ulong mysqlnd_cset_escape_slashes(const MYSQLND_CHARSET * const cset
 		char esc = '\0';
 		unsigned int len = 0;
 
-		/* check unicode characters */
-		if (cset->char_maxlen > 1 && (len = cset->mb_valid(escapestr, end))) {
+		/* check unicode characters
+		 * Encodings that have a minimum length of 1 are compatible with ASCII.
+		 * So we can skip (for performance reasons) the check to mb_valid for them. */
+		if (cset->char_maxlen > 1 && (*((zend_uchar *) escapestr) > 0x80 || cset->char_minlen > 1) && (len = cset->mb_valid(escapestr, end))) {
```

Comment:
Good idea, but we also have to check that the charset is utf8mb4. It would require breaking the public API though as `mysqlnd_cset_escape_quotes` does not take `zend_string*`. I'm not sure what impact this would have on 3rd party drivers.